src/cdk-experimental/listbox/listbox.ts

```diff
@@ -7,98 +7,207 @@
  */
 
 import {
+  AfterContentInit,
   ContentChildren,
   Directive,
-  EventEmitter, forwardRef, Inject,
-  Input, Output,
+  ElementRef, EventEmitter, forwardRef,
+  Inject,
+  Input, OnDestroy, Output,
   QueryList
 } from '@angular/core';
+import {ActiveDescendantKeyManager, Highlightable, ListKeyManagerOption} from '@angular/cdk/a11y';
+import {END, ENTER, HOME, SPACE} from '@angular/cdk/keycodes';
 import {BooleanInput, coerceBooleanProperty} from '@angular/cdk/coercion';
 
 let nextId = 0;
 
-/**
- * Directive that applies interaction patterns to an element following the aria role of option.
- * Typically meant to be placed inside a listbox. Logic handling selection, disabled state, and
- * value is built in.
- */
 @Directive({
   selector: '[cdkOption]',
   exportAs: 'cdkOption',
   host: {
     role: 'option',
     '(click)': 'toggle()',
-    '[attr.aria-selected]': 'selected || null',
+    '(focus)': 'activateOption()',
+    '(blur)': 'deactivateOption()',
+    '[attr.aria-selected]': '_selected || null',
```

Comment:
Tiny nit: I would put `[id]` above `aria-selected` so all the `attr.` bindings are together